radix-engine/src/engine/process.rs

```diff
@@ -1052,23 +1050,37 @@ impl<'r, 'l, L: SubstateStore> Process<'r, 'l, L> {
             package_address,
             blueprint_name.to_string(),
         ));
-        let result = self.call(snode_ref, function.to_string(), args);
+        let result = self.call(snode_ref, function.to_string(), args)?;
+
+        // Auto move into auth_zone
+        for proof_id in &result.proof_ids {
+            let proof = self.proofs.remove(proof_id).unwrap();
+            self.auth_zone.push(proof);
+        }
```

Comment:
This reminds me that we need special handling of buckets at transaction layer as well. Is that properly covered?

Reply:
Yes transaction layer handling of buckets is still special handled. Unifying that comes in a future PR.